### ISSUE TITLE
Evaluation reports -1 for all metrics using Mask R-CNN on Cityscapes

### ISSUE THREAD

**Issue by corrado113** [open-mmlab/mmdetection#7342]
**Describe the bug**
The Mask R-CNN model trained on Cityscapes dataset fails to report the expected metrics. According to the readme file in configs/cityscapes/README.md , the expected value for mask AP is 36.4, but the output of tools/test.py report -1 as the value of all metrics. Config and checkpoint files are those pointed in the same readme file.

```
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=1000 ] = -1.000
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=1000 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=1000 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=1000 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=1000 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=300 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=1000 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=1000 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=1000 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=1000 ] = -1.000
```
**Reproduction**

1. What command or script did you run?

```
 python tools/test.py configs/cityscapes/mask_rcnn_r50_fpn_1x_cityscapes.py checkpoints/mask_rcnn_r50_fpn_1x_cityscapes_20201211_133733-d2858245.pth --out mask_cityscapes.pkl --eval segm
```

2. Did you make any modifications on the code or config? Did you understand what you have modified?
No modifications have been performed.

3. What dataset did you use?
Cityscapes dataset, downloaded from the official website, then converted running tools/dataset_converters/cityscapes.py

**Environment**

sys.platform: linux
Python: 3.6.13 |Anaconda, Inc.| (default, Jun  4 2021, 14:25:59) [GCC 7.5.0]
CUDA available: True
GPU 0: Tesla K40m
CUDA_HOME: /usr/local/cuda-9.1
NVCC: Cuda compilation tools, release 9.1, V9.1.85
GCC: gcc (Ubuntu 5.4.0-6ubuntu1~16.04.6) 5.4.0 20160609
PyTorch: 1.5.0a0+3c31d73
PyTorch compiling details: PyTorch built with:
  - GCC 5.4
  - C++ Version: 201402
  - Intel(R) oneAPI Math Kernel Library Version 2021.4-Product Build 20210904 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v0.21.1 (Git Hash 7d2fd500bc78936d1d648ca713b901012f470dbc)
  - OpenMP 201307 (a.k.a. OpenMP 4.0)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 9.1
  - NVCC architecture flags: -gencode;arch=compute_35,code=sm_35
  - CuDNN 7.6.5  (built against CUDA 9.0)
  - Magma 2.3.0
  - Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -fopenmp -DNDEBUG -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DUSE_INTERNAL_THREADPOOL_IMPL -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format, FORCE_FALLBACK_CUDA_MPI=1, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=ON, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, USE_STATIC_DISPATCH=OFF,

TorchVision: 0.6.0a0+fe36f06
OpenCV: 4.5.5
MMCV: 1.4.4
MMCV Compiler: GCC 5.4
MMCV CUDA Compiler: 9.1
MMDetection: 2.22.0+6ddbff8

**Comment by Czm369** [comment]
Maybe you can refer to https://github.com/open-mmlab/mmdetection/blob/master/docs/en/faq.md#evaluation

**Comment by corrado113** [comment]
I believe you are referring to:

> - COCO Dataset, AP or AR = -1
> i. According to the definition of COCO dataset, the small and medium areas in an image are less than 1024 (32*32), 9216 (96*96), respectively.
> ii. If the corresponding area has no object, the result of AP and AR will set 

If that's the case, shoudn't I be able to get a score for large areas at least?
Also, what would the procedure be to obtain the mask AP score of 36.4 for Mask R-CNN, as reported in [configs/cityscapes/README.md](https://github.com/open-mmlab/mmdetection/blob/master/configs/cityscapes/README.md) ?

**Comment by chhluo** [comment]
The performance in https://github.com/open-mmlab/mmdetection/tree/master/configs/cityscapes is on val dataset.
There is no annotation in `instancesonly_filtered_gtFine_test.json`, so the results are all -1.0.
Please set `test` as following
```python
 test=dict(
      type=dataset_type,
      ann_file=data_root +
      'annotations/instancesonly_filtered_gtFine_val.json',
      img_prefix=data_root + 'leftImg8bit/val/',
      pipeline=test_pipeline)
```
